client/question/services/code_evaluators/PythonPrereqCheckServiceSpec.js

```diff
@@ -493,7 +493,70 @@ describe('PythonPrereqCheckService', function() {
       ].join('\n');
       var prereqFailureType =
         PythonPrereqCheckService.detectAndGetWrongLanguageType(code);
-      expect(prereqFailureType).toEqual(null);
+
+      expect(prereqFailureType.getErrorName()).toEqual('notOp');
+      expect(prereqFailureType.getErrorLineNumber()).toEqual(
+        prereqFailureType._errorLineNumber);
+      expect(prereqFailureType.getErrorColumnNumber()).toEqual(
+        prereqFailureType._errorColumnNumber);
+      expect(prereqFailureType._errorType).toEqual('wrongLang');
+
+    });
+  });
+
+  describe('obscureStringCharacters', function() {
+    it('obscures escaped characters', function() {
+
+      var backslash = "\\\\";
+      var backslashedBackslash = "\\\\\\\\";
+      var singlequotes = '\\\'\\\'\\\'';
+      var doublequotes = "\\\"\\\"\\\"";
+      var mixed = '  \\\' hello world\\\'s \\\'  ';
+      var newLine = '\\n';
+      var tabChar = '\\t';
+
+      expect(PythonPrereqCheckService.getObscuredCode(backslash))
+        .toEqual('xx');
+      expect(PythonPrereqCheckService.getObscuredCode(backslashedBackslash))
+        .toEqual('xxxx');
+      expect(PythonPrereqCheckService.getObscuredCode(singlequotes))
+        .toEqual('xxxxxx');
+      expect(PythonPrereqCheckService.getObscuredCode(doublequotes))
+        .toEqual('xxxxxx');
+      expect(PythonPrereqCheckService.getObscuredCode(mixed))
+        .toEqual('  xx hello worldxxs xx  ');
+      expect(PythonPrereqCheckService.getObscuredCode(newLine))
+        .toEqual('xx');
+      expect(PythonPrereqCheckService.getObscuredCode(tabChar))
+        .toEqual('xx');
+    });
+
+    it('obscures escaped characters, preserving character length', function() {
+
+      var varLine = 'test_var = \\\'1\\\' + \\\'1\\\'';
+      var obscuredVarLine = PythonPrereqCheckService.getObscuredCode(varLine);
+
+      expect(obscuredVarLine).toEqual("test_var = xx1xx + xx1xx");
+      expect(obscuredVarLine.length).toEqual(varLine.length);
+    });
+
+    it('obscure strings to test while preserving character length', function() {
+
+      var code = [
+        'def myFunction(arg):',
+        '    if "!arg":',
+        '        return arg',
+        '',
+        '    if (!arg + "abc"):',
+        '        return arg',
+        ''
+      ].join('\n');
+      var obscureCode =
+        PythonPrereqCheckService.getObscuredCode(code);
+
+      expect(code.length).toEqual(obscureCode.length);
+      expect(obscureCode).toContain('if xxxxxx');
```

Comment:
Would be a nicer test if you could just do "expect(obscureCode).toEqual(...long string...)"